### DISCUSSION TITLE
Kibana advanced settings revert to defaults.

### DISCUSSION THREAD

**Discussion by nopoxan578** [Security-Onion-Solutions/securityonion#4000] [General]
Per docs: 
_Search Results
Search results in the dashboards and through Discover are limited to the first 10 results for a particular query. If you don’t feel like this is adequate after narrowing your search, you can adjust the value for discover:sampleSize in Kibana by navigating to Stack Management -> Advanced Settings and changing the value_

However, the change does not seem to be permanent, it gets somehow reset to default value (which is 100 not 10 btw) after a while of time. The situation is same for the rest of settings.

Its either a bug or undocumented method on how those settings are managed.

**Comment by defensivedepth** [comment]
Confirmed with `.40` that when a salt highstate runs (every 15min), the value is changed to the default value of 100.

I have created an issue to track this: https://github.com/Security-Onion-Solutions/securityonion/issues/3933

**Comment by nopoxan578** [comment]
Well, In my case it was the situation for all kibana advanced settings, like custom time picker periods, and discoverSamplesize, etc..

**Comment by nopoxan578** [comment]
Also field formatting from index management in kibana settings.. any changes are reverted back each salt highstate..
